xds/src/main/java/io/grpc/xds/ClusterImplLoadBalancer.java

```diff
@@ -207,7 +208,36 @@ public Subchannel createSubchannel(CreateSubchannelArgs args) {
         addresses.add(new EquivalentAddressGroup(eag.getAddresses(), attrBuilder.build()));
       }
       args = args.toBuilder().setAddresses(addresses).build();
-      return delegate().createSubchannel(args);
+      final Subchannel subchannel = delegate().createSubchannel(args);
+      Locality locality = subchannel.getAllAddresses().get(0).getAttributes().get(
+          InternalXdsAttributes.ATTR_LOCALITY);  // all addresses should be in the same locality
+      // Endpoint addresses resolved by ClusterResolverLoadBalancer should always contain
+      // attributes with its locality, including endpoints in LOGICAL_DNS clusters.
+      // In case of not (which really shouldn't), loads are aggregated under an empty locality.
+      if (locality == null) {
+        locality = new Locality("", "", "");
+      }
+      final ClusterLocalityStats localityStats = xdsClient.addClusterLocalityStats(
+          cluster, edsServiceName, locality);
+
+      return new ForwardingSubchannel() {
+        @Override
+        public Attributes getAttributes() {
+          return delegate().getAttributes().toBuilder().set(
```

Comment:
Put stats object in PickSubchannelArgs instead as being pointed out offline.

Reply:
Done.